### ISSUE TITLE
Negative numbers not being invalidated

### ISSUE THREAD

**Issue by ivansg44** [Public-Health-Bioinformatics/covid19ValidationGrid#34]
Fields tagged as `"decimal"` and `"integer"` allow negative values. This does not make sense for any of the current fields.

In addition, the reference guide for several of these fields tells users to insert a generic "numerical value". I don't know if its common sense, but stating "positive numerical value" would be more accurate. "Positive integer value" would be even more accurate for integer fields.

`"decimal"` and `"integer"` are also not be the best `datatype` values if we only allow positive values. Wikipedia has a list of descriptors for sets of numbers: https://en.wikipedia.org/wiki/List_of_types_of_numbers. Possible improvements on `"integer"` are `"whole numbers"`, `"natural numbers"`, and `"non-negative integers"`. Possible improvements on `"decimal"` are `"positive numbers"` or `"positive decimals"`. **However, I don't believe the users ever see the `datatype` values, so this is non-important from a ui point-of-view.**

**Comment by ddooley** [comment]
I would say we can switch to the xml data types if you want now.  https://www.w3schools.com/xml/schema_dtypes_numeric.asp
Also detailed in https://www.w3.org/TR/xmlschema-2/type-hierarchy.gif 
from https://www.w3.org/TR/xmlschema-2

That keeps the ship heading in the direction of schema.org and/or owl ontology control of data types without having to reinvent stuff.

There are a number of other bells and whistles for numeric validation that we would eventually add to the validator.  Upper and lower bounds, out of range, degrees of precision.  Start to get into business of silent conversion – e.g. from 1.0 to 1.  Rounding up and down.  Thought I’d mention GEEM will be providing both schema.org and native owl parameters for controlling all that.  

In this particular app I'd venture that its not really important to catch negative numbers as that’s not something that happens as a result of manual data entry error.

**Comment by ivansg44** [comment]
I like the idea of using xml data types, except that there doesn't seem to be a non-negative decimal. However, we could use lower bounds to deal with that. Could you take the lead on switching to xml data types?

> In this particular app I'd venture that its not really important to catch negative numbers as that’s not something that happens as a result of manual data entry error.

I agree that it will not be a common manual entry error, but disagree that it's not important to catch, because typing `"foo"` or any other non-number into a decimal field is not going to be a common manual entry error either, but we still have logic to catch non-numbers in decimal fields.

**Comment by ddooley** [comment]
Sure, I can take on some adjustments for more precise xsd:decimal, etc. types.  Some validation that's at bottom of geem_form.js may be reusable.  Shall we include min/max values now then too?

**Comment by ivansg44** [comment]
Yes, let's do the whole shebang. I'll take a look at `geem_form.js` after you update `data.js`.
